### ISSUE TITLE
Feature Request: $mapper->reset() to optionally also clear the virtual fields

### ISSUE THREAD

**Issue by ethanpil** [bcosca/fatfree#667]
I propose we add an optional parameter to `$mapper->reset()` that will also remove all of the vfields.

```
 function $mapper->reset($vfields = False)  {
```

Per https://groups.google.com/forum/#!topic/f3-framework/RQA-NOcuBWk

**Comment by jimymodi** [comment]
@ethanpil What do you suggest the default behavior should be `$mapper->reset($vfields = False)  {` or `$mapper->reset($vfields = False)  {` . So by default virtual fields should be wiped out or done only on request.

**Comment by ethanpil** [comment]
I think we will do best to keep the default function the same to ensure
backwards compatibility.

**Comment by Rayne** [comment]
@jimymodi  created PR https://github.com/bcosca/fatfree/pull/835 for this issue.

**Comment by Rayne** [comment]
I don't like the idea to add virtual fields and then drop them later to reuse the object. This could cause unexpected side effects. At least the default value `$vfields = false` is reasonable.

**Comment by xfra35** [comment]
I don't think it should belong to `reset()`, because this method is used to reset the cursor, not to modify the mapper structure.

Why not adding an `adhoc()` method, so we can loop and clear all adhoc fields?

```php
foreach ($mapper->adhoc() as $key)
  $mapper->clear($key);
```
This solution also allows to filter adhoc fields out:
```php
foreach ($mapper->adhoc() as $key)
  if (preg_match('/^total_/',$key))
    $mapper->clear($key);
```

**Comment by ethanpil** [comment]
How about `$mapper->wipe()` or `$mapper->reboot()` or something similar which can be used to simply restore the mapper to brand new status.
